aiida_mlip/calculations/singlepoint.py

```diff
@@ -0,0 +1,271 @@
+"""Class to run single point calculations."""
+
+from aiida.common import datastructures
+import aiida.common.folders
+from aiida.engine import CalcJob, CalcJobProcessSpec
+import aiida.engine.processes
+from aiida.orm import Dict, SinglefileData, Str, StructureData
+
+from aiida_mlip.data.model import ModelData
+
+
+class Singlepoint(CalcJob):  # numpydoc ignore=PR01
+    """
+    Calcjob implementation to run single point calculations using mlips.
+
+    Attributes
+    ----------
+    _DEFAULT_OUTPUT_FILE : str
+        Default stdout file name.
+    _DEFAULT_INPUT_FILE : str
+        Default input file name.
+    _XYZ_OUTPUT : str
+        Default xyz output file name.
+    _LOG_FILE : str
+        Default log file name.
+
+    Methods
+    -------
+    define(spec: CalcJobProcessSpec) -> None:
+        Define the process specification, its inputs, outputs and exit codes.
+    validate_inputs(value: dict, port_namespace: PortNamespace) -> Optional[str]:
+        Check if the inputs are valid.
+    prepare_for_submission(folder: Folder) -> CalcInfo:
+        Create the input files for the `Calcjob`.
+    """
+
+    _DEFAULT_OUTPUT_FILE = "aiida-stdout.txt"
+    _DEFAULT_INPUT_FILE = "aiida.cif"
+    _XYZ_OUTPUT = "aiida-results.xyz"
+    _LOG_FILE = "aiida.log"
+
+    @classmethod
+    def define(cls, spec: CalcJobProcessSpec):
+        """
+        Define the process specification, its inputs, outputs and exit codes.
+
+        Parameters
+        ----------
+        spec : `aiida.engine.CalcJobProcessSpec`
+            The calculation job process spec to define.
+        """
+        super().define(spec)
+
+        # Define inputs
+        spec.input(
+            "calctype",
+            valid_type=Str,
+            default=lambda: Str("singlepoint"),
+            help="calculation type (single point or geom opt)",
+        )
+        spec.input(
+            "architecture",
+            valid_type=Str,
+            default=lambda: Str("mace_mp"),
+            help="Architecture to use for calculation, defaults to mace_mp",
+        )
+        spec.input(
+            "model",
+            valid_type=ModelData,
+            required=False,
+            help="mlip model used for calculation",
+        )
+        spec.input("structure", valid_type=StructureData, help="The input structure.")
+        spec.input("precision", valid_type=Str, help="Precision level for calculation")
+        spec.input(
+            "device",
+            valid_type=Str,
+            required=False,
+            default=lambda: Str("cpu"),
+            help="Device in which to run calculation(cpu, gpu...)",
```

Comment:
```suggestion
            help="Device on which to run calculation (cpu, cuda or mps)",
```
Probably misleading to say "gpu" anyway